plugins/flytekit-ray/flytekitplugins/ray/models.py

```diff
@@ -1,10 +1,170 @@
+import json as _json
 import typing
 
+from flyteidl.core import tasks_pb2 as _core_task
 from flyteidl.plugins import ray_pb2 as _ray_pb2
+from google.protobuf import json_format as _json_format
+from google.protobuf import struct_pb2 as _struct
 
 from flytekit.models import common as _common
 
 
+class K8sObjectMetadata(_common.FlyteIdlEntity):
+    def __init__(self, labels: typing.Dict[str, str] = None, annotations: typing.Dict[str, str] = None):
+        """
+        This defines additional metadata for building a kubernetes pod.
+        """
+        self._labels = labels
+        self._annotations = annotations
+
+    @property
+    def labels(self) -> typing.Dict[str, str]:
+        return self._labels
+
+    @property
+    def annotations(self) -> typing.Dict[str, str]:
+        return self._annotations
+
+    def to_flyte_idl(self) -> _core_task.K8sObjectMetadata:
+        return _core_task.K8sObjectMetadata(
+            labels={k: v for k, v in self.labels.items()} if self.labels is not None else None,
+            annotations={k: v for k, v in self.annotations.items()} if self.annotations is not None else None,
+        )
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object: _core_task.K8sObjectMetadata):
+        return cls(
+            labels={k: v for k, v in pb2_object.labels.items()} if pb2_object.labels is not None else None,
+            annotations={k: v for k, v in pb2_object.annotations.items()}
+            if pb2_object.annotations is not None
+            else None,
+        )
+
+
+class IOStrategy(_common.FlyteIdlEntity):
+    """
+    Provides methods to manage data in and out of the Raw container using Download Modes. This can only be used if DataLoadingConfig is enabled.
+    """
+
+    DOWNLOAD_MODE_EAGER = _core_task.IOStrategy.DOWNLOAD_EAGER
+    DOWNLOAD_MODE_STREAM = _core_task.IOStrategy.DOWNLOAD_STREAM
+    DOWNLOAD_MODE_NO_DOWNLOAD = _core_task.IOStrategy.DO_NOT_DOWNLOAD
+
+    UPLOAD_MODE_EAGER = _core_task.IOStrategy.UPLOAD_EAGER
+    UPLOAD_MODE_ON_EXIT = _core_task.IOStrategy.UPLOAD_ON_EXIT
+    UPLOAD_MODE_NO_UPLOAD = _core_task.IOStrategy.DO_NOT_UPLOAD
+
+    def __init__(
+        self,
+        download_mode: _core_task.IOStrategy.DownloadMode = DOWNLOAD_MODE_EAGER,
+        upload_mode: _core_task.IOStrategy.UploadMode = UPLOAD_MODE_ON_EXIT,
+    ):
+        self._download_mode = download_mode
+        self._upload_mode = upload_mode
+
+    def to_flyte_idl(self) -> _core_task.IOStrategy:
+        return _core_task.IOStrategy(download_mode=self._download_mode, upload_mode=self._upload_mode)
+
+    @classmethod
+    def from_flyte_idl(cls, pb2_object: _core_task.IOStrategy):
+        if pb2_object is None:
+            return None
+        return cls(
+            download_mode=pb2_object.download_mode,
+            upload_mode=pb2_object.upload_mode,
+        )
+
+
+class DataLoadingConfig(_common.FlyteIdlEntity):
```

Comment:
Do we think we'd need this for the ray plugin?

Reply:
we do not